nengo/builder/ensemble.py

```diff
@@ -88,6 +88,13 @@ def get_gain_bias(ens, rng=np.random):
         bias = get_samples(ens.bias, ens.n_neurons, rng=rng)
         max_rates, intercepts = ens.neuron_type.max_rates_intercepts(
             gain, bias)
+        if (ens.max_rates is not Ensemble.max_rates.default or
+           ens.intercepts is not Ensemble.intercepts.default):
+            warnings.warn(
+                "Specifying the gains and biases for %s imposes a set of "
+                "maximum firing rates and intercepts. Further specifying "
+                "either max_rates or intercepts has no effect." % ens)
```

Comment:
I would make this a `NengoWarning`.

Reply:
When initially thinking about this, I stuck with a `UserWarning` for consistency with the warning issued by `gen_eval_points`. In any event, I've added a fixup to use a `NengoWarning`, but it doesn't seem like there is a widespread usage of these throughout the codebase (i.e. most warnings are `UserWarning`s based on a quick search). Is there a rule of thumb for when one should preferred over the other? Or should most of the existing `UserWarning`s be switched over to `NengoWarning`s? (I'm happy to do this at some point if that is the case)